client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -304,7 +305,7 @@ public CompletionStage<ServerStatistics> serverStatisticsAsync() {
 
    @Override
    public K keyAsObjectIfNeeded(Object key) {
```

Comment:
I removed it from all the other places. The streaming uses it because it never passes the data format when instantiating the commands :/ Otherwise, we would need to change all the streaming commands to include the argument in the constructor.